### PR TITLE
added parser for Hindawi

### PR DESCRIPTION
**Pull request by librarywebchic**

### CODE REVIEW

File: hdwi/parser.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  let match;
+
+  if ((match = /^\/journals\/([a-z]+)\/([0-9]{4})\/([0-9]+)\/$/i.exec(path)) !== null) {
+    // https://www.hindawi.com/journals/apec/2021/5582774/
+    result.rtype    = 'ARTICLE';
+    result.mime     = 'HTML';
+    result.title_id = match[1];
+    result.publication_date = match[2]
+    result.unitid = match[3];
+    result.doi = match[3];
```
Comment (felixleo22) [inline hdwi/parser.js:31]: this id is not a DOI, please change the analyse in analogist

File: hdwi/parser.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  } else if ((match = /^\/journals\/([a-z]+)\/([0-9]{4})\/([0-9]+)\.ris$/i.exec(path)) !== null) {
+    // https://www.hindawi.com/journals/apec/2021/5582774.ris
+    result.rtype    = 'RECORD_VIEW';
+    result.mime     = 'RIS';
+    result.title_id = match[1];
+    result.publication_date = match[2]
+    result.unitid = match[3];
```
Comment (felixleo22) [inline hdwi/parser.js:39]: this id is not a DOI, please change the analyse in analogist

File: hdwi/parser.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    result.doi = match[3];
+
+  } else if ((match = /^\/journals\/([a-z]+)\/([0-9]{4})\/([0-9]+)\.ris$/i.exec(path)) !== null) {
+    // https://www.hindawi.com/journals/apec/2021/5582774.ris
+    result.rtype    = 'RECORD_VIEW';
+    result.mime     = 'RIS';
+    result.title_id = match[1];
+    result.publication_date = match[2]
```
Comment (felixleo22) [inline hdwi/parser.js:38]:
```suggestion
    result.publication_date = match[2];
```

File: hdwi/parser.js (hunk 1 above)
Comment (wilmouths) [inline hdwi/parser.js:31]: This ID is not a DOI, after investigations on the platform, it seems that the DOI is : ``10.1155/2021/5582774``
Please add a DOI prefix (``const doiPrefix = '10.1155';``) as explained on a previous parser, modify the analogist analysis.

Example : 
image

File: hdwi/parser.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    result.unitid = match[3];
+    result.doi = match[3];
```
Comment (wilmouths) [inline hdwi/parser.js:40]: This ID is not a DOI, after investigations on the platform, it seems that the DOI is : ``10.1155/2021/5582774``
Please add a DOI prefix (``const doiPrefix = '10.1155';``) as explained on a previous parser, modify the analogist analysis.

Example : 
image

**Comment by wilmouths** [comment]
@librarywebchic 

I updated your parser and analyses on Analogist, added the DOI prefix and changed the value of DOI and unit.
Pouvez-vous mettre à jour votre dépôt basé sur le notre, une mise à jour des tests a été faite.

**Pull request merged**